src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -107,7 +107,7 @@ SHAMap::getStack (uint256 const& id, bool include_nonmatching_leaf) const
     }
 
     if (include_nonmatching_leaf ||
-        (std::static_pointer_cast<SHAMapTreeNode>(node)->peekItem()->getTag() == id))
+        (std::static_pointer_cast<SHAMapTreeNode>(node)->peekItem()->key() == id))
```

Comment:
`node` can't be `nullptr` here else we would have crashed in the loop right above this code.
Given a non-`nullptr` input, `std::static_pointer_cast` can't return `nullptr`.  `std::dynamic_pointer_cast` is another matter:  it *can* return `nullptr`.